### PR TITLE
Add git url as parameter

### LINKED ISSUE #54: [Feature request]: Add support for selfhosted git
### Is your feature request related to a problem? Please describe.

I have a selfhosted Gitea instance. Instead of backing up to github, I would like to backup to my gitea instance. 

### Describe the solution you'd like

After doing some testing with the script, the only thing that need to be changed is the github.com domain here:
https://github.com/Staubgeborener/klipper-backup/blob/main/script.sh#L57
and here:
https://github.com/Staubgeborener/klipper-backup/blob/main/script.sh#L62

If you could also change this into a variable which defaults to github.com this should be easily integratable. 

### Describe alternatives you've considered

...

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by TheZoker**
A sample integration of the git URL parameter

Closes #54

### CODE REVIEW

**Review by Tylerjet — CHANGES_REQUESTED** [review]
I think a better option, just to ensure it is backwards compatible is to revert to default if the variable in .env is not defined.

Ex.) at line 11 of script.sh adding git_url=${git_url:-"github.com"} 

this should ensure that if git_url is not defined it would still know to continue using github.com

This makes git_url more of an optional addition into .env instead of being required

**Comment by Staubgeborener** [comment]
I have the same opinion as @Tylerjet. Could you please change your PR (remove the line in `.env` file and adjust the corresponding lines in `script.sh`). After that we can merge that and change the docs for this optional parameter.

**Comment by TheZoker** [comment]
I implemented your suggestions :)
Thanks for the review!

**Comment by Staubgeborener** [comment]
I started a review, please check this.

Also, we implemented some if-statements (like [this](https://github.com/Staubgeborener/klipper-backup/blob/main/script.sh#L55) or [this](https://github.com/Staubgeborener/klipper-backup/blob/main/script.sh#L60)) in order to update the `.git` folder inside the `config_backup` folder and by doing so be more flexible. Otherwise, the user will get an error in the following example scenario:

1. add `git_url=some-url.com` in `.env`
2. run `./script.sh` in order to push files
3. edit `git_url=some-other-url.com` in `.env`
4. run `./script.sh` again in order to push files

In this case, git still tries to push the files to `some-url.com` as the URL is not updated by itself. The if-statements check if the corresponding line inside the `.env` file does not match with the information inside the `~/config_backup/.git` folder. So we need something like

`if [[ "$git_url" != $(git remote get-url origin | sed command to grep the some-url.com url) ]];`

And right now as i write this stuff here: This reminds me of another issue. If we do so, [line 61](https://github.com/Staubgeborener/klipper-backup/blob/main/script.sh#L61) won't work anymore as `github.com` is hard-coded there, so also need to change this line.

**Comment by TheZoker** [comment]
Can't we just set the URL on every run like this?
```
git remote set-url origin $git_url
```

That way we don't have to check if the URL is different, we just always use the current set URL

File: script.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ source "$parent_path"/.env
 
 backup_folder="config_backup"
 backup_path="$HOME/$backup_folder"
+git_url=${git_url:-"github.com"}
```
Comment (Tylerjet) [inline script.sh:11]:
```suggestion
git_host=${git_host:-"github.com"}
full_git_url="https://"$github_token"@"$git_host"/"$github_username"/"$github_repository".git"
```

File: script.sh [hunk 2]
```diff
@@ -54,12 +54,12 @@ fi
 
 # Check if remote origin already exists and create if one does not
 if [ -z "$(git remote get-url origin 2>/dev/null)" ]; then
-    git remote add origin https://"$github_token"@github.com/"$github_username"/"$github_repository".git
+    git remote add origin https://"$github_token"@"$git_url"/"$github_username"/"$github_repository".git
```
Comment (Tylerjet) [inline script.sh:58]:
```suggestion
    git remote add origin "$full_git_url"
```

File: script.sh [hunk 3]
```diff
@@ -54,12 +54,12 @@ fi
...
 fi
 
 # Check if remote origin changed and update when it is
 if [[ "$github_repository" != $(git remote get-url origin | sed 's/https:\/\/.*@github.com\///' | sed 's/\.git$//' | xargs basename) ]]; then
-    git remote set-url origin https://"$github_token"@github.com/"$github_username"/"$github_repository".git
+    git remote set-url origin https://"$github_token"@"$git_url"/"$github_username"/"$github_repository".git
```
Comment (Tylerjet) [inline script.sh:63]:
```suggestion
if [[ "$full_git_url" != $(git remote get-url origin) ]]; then
    git remote set-url origin "$full_git_url"
fi
```

**Review by Tylerjet — CHANGES_REQUESTED** [review]
was just talking with @Staubgeborener about it and likely makes sense to also generate a full_git_url to use throughout the script, added the suggestions to the following lines this ensures we are checking against the full repo url, and only need to change the url in one location instead of 3 or more in the future. also renamed git_url to git_host since its not really a full url but more of a host/hostname

If needed i can also submit a pr with the above changes as i was going to be pushing one already with a few other changes and can merge this in.

**Comment by Tylerjet** [comment]
ah that works as well hehe

**Comment by TheZoker** [comment]
I pushed you changes 👍

**Comment by Tylerjet** [comment]
@TheZoker one last thing actually is the final lines from line ~61 - 64

```
# Check if remote origin changed and update when it is https://"$github_token"@"$git_url"/"$github_username"/"$github_repository".git
if [[ "$full_git_url" != $(git remote get-url origin) ]]; then
    git remote set-url origin "$full_git_url"
fi
fi
```

it should be 
```
# Check if remote origin changed and update when it is
if [[ "$full_git_url" != $(git remote get-url origin) ]]; then
    git remote set-url origin "$full_git_url"
fi
```

for some reason my suggestion didn't substitute the right section

File: script.sh [hunk 4]
```diff
@@ -55,12 +56,14 @@ fi
 
 # Check if remote origin already exists and create if one does not
 if [ -z "$(git remote get-url origin 2>/dev/null)" ]; then
-    git remote add origin https://"$github_token"@"$git_url"/"$github_username"/"$github_repository".git
+    git remote add origin "$full_git_url"
 fi
 
 # Check if remote origin changed and update when it is
 if [[ "$github_repository" != $(git remote get-url origin | sed 's/https:\/\/.*@github.com\///' | sed 's/\.git$//' | xargs basename) ]]; then
```
Comment (Tylerjet) [inline script.sh:63]:
```suggestion
```

File: script.sh [hunk 5]
```diff
@@ -55,12 +56,14 @@ fi
...
-    git remote set-url origin https://"$github_token"@"$git_url"/"$github_username"/"$github_repository".git
+if [[ "$full_git_url" != $(git remote get-url origin) ]]; then
+    git remote set-url origin "$full_git_url"
+fi
 fi
```
Comment (Tylerjet) [inline script.sh:67]:
```suggestion
```

**Review by Tylerjet — CHANGES_REQUESTED** [review]
this should fix it

**Comment by Tylerjet** [comment]
Perfect!

**Comment by TheZoker** [comment]
I just saw that and was already fixing it, but this way it was faster :D

**Comment by Tylerjet** [comment]
@Staubgeborener should be good to merge now!

**Comment by Staubgeborener** [comment]
Nice work!

**Pull request merged**